### PR TITLE
Agriculture production consumption trade

### PR DESCRIPTION
**Pull request by aabdaab**
This PR adds Bar Chart for presenting meat consumption, production and trade. It's added to Country Context section for Agriculture page.

image

**TO DO**
Stacked data for other countries is yet to be added as I'm waiting for this task to be done: https://www.pivotaltracker.com/story/show/163569429

### CODE REVIEW

**Comment by kfilip94** [comment]
Hey Marta, I checked this :meat_on_bone: section and it looks great :+1:, just few things to fix below:
- Is it possible to set a white background for filter dropdowns and legend elements?
- Could you change content padding, marked below? I think these spaces are a bit different on the design
screenshot from 2019-01-30 14-32-51
- Also I think you can improve width scaling of filter dropdowns, especially for lower resolutions because  categories list is almost unreadable
screenshot from 2019-01-30 14-49-16
I'll check the code now

File: app/javascript/app/components/sectors-agriculture/countries-context/meat-data/tooltip/tooltip-component.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  const countryColor = get(config, 'theme.yCountry.stroke') || '';
+  const othersColor = get(config, 'theme.yOthers.stroke') || '';
+  const unit = get(config, 'axes.yLeft.unit') || '';
+
+  return (
+    <div className={styles.tooltip}>
+      <div className={styles.category}>{category}</div>
+      <p className={styles.info}>
+        <Tag
```
Comment (kfilip94) [inline app/javascript/app/components/sectors-agriculture/countries-context/meat-data/tooltip/tooltip-component.jsx]: I think you can extract <Tag /> component to separate method `renderTag()` and pass name, value and color properties.

File: app/javascript/app/components/sectors-agriculture/countries-context/meat-data/meat-data-component.jsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+              />
+            )}
+          </div>
+        </div>
+        <div className={styles.yearData}>
+          <span>{selectedYear && selectedYear.value}</span> data
+        </div>
+      </Card>
+    )}
+    {selectedCountry &&
```
Comment (kfilip94) [inline app/javascript/app/components/sectors-agriculture/countries-context/meat-data/meat-data-component.jsx]: I think you can save params as a one object (before `return (...)`):
`const params = selectedCountry && selectedYear && { country: selectedCountry.value, year: selectedYear.value };`
and then use `params` object as a guard here:
`{params && [ <MeatConsumptionProvider... ]}`

**Comment by simaob** [comment]
Ready again?

**Comment by kfilip94** [comment]
I'm checking this chart again and I think I found something worth to discuss:
- most of the countries here have a marginal contribution to global meat production/consumption/trade:
screenshot from 2019-01-31 11-58-03
We don't see here any data for selected country because it's too small to be visible on this chart in relation to other countries.
So in most of the cases bar chart doesn't show us valuable information, only tooltip does.
- Also there are cases when we don't have a data for selected country at all, so maybe we should set 'no data available' message instead of showing bar chart without country data?
screenshot from 2019-01-31 12-13-45

**Comment by kfilip94** [comment]
Marta, thanks for changes! :+1: For me it looks good, we can discuss this comment above later

**Pull request merged**